metric_tank/aggmetric.go

```diff
@@ -70,10 +72,11 @@ func (a *AggMetric) GrowNumChunks(numChunks uint32) {
 // it optionally also creates aggregations with the given settings
 func NewAggMetric(key string, chunkSpan, numChunks uint32, aggsetting ...aggSetting) *AggMetric {
 	m := AggMetric{
-		Key:       key,
-		ChunkSpan: chunkSpan,
-		NumChunks: numChunks,
-		Chunks:    make([]*Chunk, 0, numChunks),
+		Key:        key,
+		ChunkSpan:  chunkSpan,
+		NumChunks:  numChunks,
+		Chunks:     make([]*Chunk, 0, numChunks),
+		writeQueue: make(chan *Chunk, *maxUnwrittenChunks),
```

Comment:
can we cleanly pass the maxUnwrittenChunks option through, instead of depending on global variables.